sklearn/multilabel/classifier_chain.py

```diff
@@ -0,0 +1,134 @@
+import numpy as np
+from ..base import BaseEstimator, clone
+from ..utils import check_random_state
+from scipy.sparse import issparse
+
+
+class ClassifierChain(BaseEstimator):
+    """
+    A multi-label model that arranges binary classifiers into a chain. Each
+    model makes a prediction in the order specified by the chain using all
+    of the available features provided to the model plus the predictions of
+    models that are earlier in the chain.
+
+    By default the order of the chain is random although it can be specified
+    at the time of fitting with the chain_order parameter. Since the optimal
+    chain_order is not known a priori it is common to use the mean prediction
+    of an ensemble randomly ordered classifier chains.
+
+    References
+    ----------
+    Jesse Read, Bernhard Pfahringer, Geoff Holmes, Eibe Frank, "Classifier
+    Chains for Multi-label Classification", 2009.
+
+    """
+
+    def __init__(self, base_estimator, random_state=None,
+                 order=None, shuffle=True):
+        """
+        Parameters
+        ----------
+        base_estimator : estimator
+            The base estimator from which the classifier chain is built.
+
+        random_state : int or RandomState, optional, default None
+            State or seed for random number generator.
+
+        order : list
```

Comment:
make this an `array-like, shape (n_outputs,)`

and add tests to ensure that list/array are both supported.